--- a/indico/htdocs/js/jquery/jquery-ui.js
+++ b/indico/htdocs/js/jquery/jquery-ui.js
@@ -6902,6 +6902,7 @@ function Datepicker() {
 		constrainInput: true, // The input is constrained by the current date format
 		showButtonPanel: false, // True to show button panel, false to not show it
 		autoSize: false, // True to size the input for the date format, false to leave as is
+		unifyNumRows: false, // True to always use six rows; ensuring datepickers showing different months having the same height
 		disabled: false // The initial disabled state
 	};
 	$.extend(this._defaults, this.regional[""]);
@@ -8379,7 +8380,7 @@ $.extend(Datepicker.prototype, {
 	_generateHTML: function(inst) {
 		var maxDraw, prevText, prev, nextText, next, currentText, gotoDate,
 			controls, buttonPanel, firstDay, showWeek, dayNames, dayNamesMin,
-			monthNames, monthNamesShort, beforeShowDay, showOtherMonths,
+			monthNames, monthNamesShort, beforeShowDay, showOtherMonths, unifyNumRows,
 			selectOtherMonths, defaultDate, html, dow, row, group, col, selectedDate,
 			cornerClass, calender, thead, day, daysInMonth, leadDays, curRows, numRows,
 			printDate, dRow, tbody, daySettings, otherMonth, unselectable,
@@ -8464,6 +8465,7 @@ $.extend(Datepicker.prototype, {
 		showOtherMonths = this._get(inst, "showOtherMonths");
 		selectOtherMonths = this._get(inst, "selectOtherMonths");
 		defaultDate = this._getDefaultDate(inst);
+		unifyNumRows = this._get(inst, 'unifyNumRows');
 		html = "";
 		dow;
 		for (row = 0; row < numMonths[0]; row++) {
@@ -8505,7 +8507,7 @@ $.extend(Datepicker.prototype, {
 					inst.selectedDay = Math.min(inst.selectedDay, daysInMonth);
 				}
 				leadDays = (this._getFirstDayOfMonth(drawYear, drawMonth) - firstDay + 7) % 7;
-				curRows = Math.ceil((leadDays + daysInMonth) / 7); // calculate the number of rows to generate
+				curRows = unifyNumRows ? 6 : Math.ceil((leadDays + daysInMonth) / 7); // calculate the number of rows to generate
 				numRows = (isMultiMonth ? this.maxRows > curRows ? this.maxRows : curRows : curRows); //If multiple months, use the higher number of rows (see #7043)
 				this.maxRows = numRows;
 				printDate = this._daylightSavingAdjust(new Date(drawYear, drawMonth, 1 - leadDays));
@@ -9160,7 +9160,7 @@ $.widget( "ui.dialog", {
                        }
                });

-           this.uiDialogTitlebarClose = $("<button></button>")
+           this.uiDialogTitlebarClose = $("<button type='button'></button>")
                        .button({
                                label: this.options.closeText,
                                icons: {
